File: backend/src/main/java/br/com/videolocadorapassatempo/model/ItemModel.java
package br.com.videolocadorapassatempo.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;

@Entity
@Table(name = "item")
@Getter
@Setter
public class ItemModel implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequence_item")
    @SequenceGenerator(name = "sequence_item", sequenceName = "sequence_item", allocationSize = 1)
    private Long id;

    @Column(name = "serial_number", nullable = false)
    private String serialNumber;

    @Column(name = "purchase_date", nullable = false)
    private LocalDate purchaseDate;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_title", referencedColumnName = "id", nullable = false)
    private TitleModel titleModel;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_item_type", nullable = false)
    private ItemTypeModel itemTypeModel;

}

File: frontend/src/app/shared/models/item.model.ts
import { Title } from './title.model';
import { BasicModel } from './basic.model';
import { ItemType } from './item-type.model';

export class Item extends BasicModel {
  constructor(
    id?: number,
    public serialNumber?: string,
    public titleModel?: Title,
    public purchaseDate?: string,
    public itemTypeModel?: ItemType,
  ) {
    super(id);
  }
}